packages/samples/headless-ssr/cypress/e2e/ssr-search-parameter-manager.cy.ts

```diff
@@ -0,0 +1,160 @@
+import {ConsoleAliases, spyOnConsole, waitForHydration} from './ssr-e2e-utils';
+
+const searchStateKey = 'search-state';
+
+describe('headless ssr with search parameter manager example', () => {
+  describe('when loading a page without search parameters, after hydration', () => {
+    beforeEach(() => {
+      spyOnConsole();
+      cy.visit('/');
+      waitForHydration();
+    });
+
+    it('should not update the search parameters', () => {
+      cy.url().should((href) =>
+        expect(new URL(href).searchParams.size).to.equal(0)
+      );
+    });
+
+    describe('after submitting a search', () => {
+      const query = 'abc';
+      beforeEach(() =>
+        cy.get('.search-box input').focus().type(`${query}{enter}`)
+      );
+
+      describe('after the url was updated', () => {
+        beforeEach(() =>
+          cy
+            .url()
+            .should((href) =>
+              expect(new URL(href).searchParams.has(searchStateKey)).to.equal(
+                true
+              )
+            )
+        );
+
+        it('should have the correct search parameters', () => {
+          cy.url().should((href) => {
+            const searchState = new URL(href).searchParams.get(searchStateKey);
+            expect(searchState && JSON.parse(searchState)).to.deep.equal({
+              q: query,
+            });
+          });
+        });
+
+        it('only has two history states', () => {
```

Comment:
What is this testing? Is it dependent on previous tests?

Reply:
It's testing that we didn't do `history.push` multiple times, which could happen if:
* The listener of `searchParametersManager.subscribe` is called twice.
* The hook we use to push new history states is something like `useEffect` and has too many dependencies, causing it to mistakenly fire more times than necessary.